### PR TITLE
Implement secrets via bind-mounts for local compose

### PR DESCRIPTION
**Pull request by aiordache**
Implement compose secrets via bind mounts. 
 
```yaml
services:
  test:
    image: nginx
    secrets:
      - my_secret
secrets:
  my_secret:
    file: ./my_secret.txt
```
```shell
$ cat my_secret.txt 
blabla bbla ba
```
```
docker compose up -d
The new 'docker compose' command is currently experimental. To provide feedback or request new features please open issues at https://github.com/docker/compose-cli
[+] Running 2/1
 ⠿ Network "test_default"  Created                                                               0.5s
 ⠿ test_test_1             Created                                                               0.0s

```
```
$ docker exec -it test_test_1 cat /run/secrets/my_secret
blabla bbla ba

```

Closes https://github.com/docker/compose-cli/issues/934

### CODE REVIEW

File: local/compose/create.go [hunk 1]
```diff
@@ -296,6 +296,32 @@ func buildContainerMountOptions(p types.Project, s types.ServiceConfig, inherit
...
+		}
+		target := "/var/run/secrets"
+		if secret.Target == "" {
+			target = fmt.Sprintf("%s/%s", target, secret.Source)
+		} else if !filepath.IsAbs(secret.Target) {
+			target = fmt.Sprintf("%s/%s", target, secret.Target)
+		}
+		defSecret, ok := p.Secrets[secret.Source]
+
+		if !ok {
```
Comment (ndeloof) [inline local/compose/create.go:311]: this is already checked by compose-go loader
https://github.com/compose-spec/compose-go/blob/master/loader/validate.go#L51

File: local/compose/create.go [hunk 2]
```diff
@@ -296,6 +296,32 @@ func buildContainerMountOptions(p types.Project, s types.ServiceConfig, inherit
 		}
 		mounts = append(mounts, mount)
 	}
+	for _, secret := range s.Secrets {
+		if contains(inherited, secret.Target) {
```
Comment (ndeloof) [inline local/compose/create.go:300]: doing so, IIUC, will prevent changing the secret file name in compose file and get sercet bind updated on `up`.
IMHO inherited volumes from previous container should be default be overriden by config.
Not sure how docker-compose handle this, we should align

File: local/compose/create.go [hunk 3]
```diff
@@ -296,6 +296,32 @@ func buildContainerMountOptions(p types.Project, s types.ServiceConfig, inherit
...
+			target = fmt.Sprintf("%s/%s", target, secret.Source)
+		} else if !filepath.IsAbs(secret.Target) {
+			target = fmt.Sprintf("%s/%s", target, secret.Target)
+		}
+		defSecret, ok := p.Secrets[secret.Source]
+
+		if !ok {
+			return nil, fmt.Errorf("secret %s is not defined", secret.Source)
+		}
+		mount, err := buildMount(p, types.ServiceVolumeConfig{
```
Comment (ndeloof) [inline local/compose/create.go:314]: should first check this is _not_ `External`

**Review by ndeloof — CHANGES_REQUESTED** [review]
some redundant code with compose-go and a potential distinct behaviour vs docker-compose to be confirmed

File: local/compose/create.go [hunk 4]
```diff
@@ -296,6 +296,30 @@ func buildContainerMountOptions(p types.Project, s types.ServiceConfig, inherit
 		}
 		mounts = append(mounts, mount)
 	}
+	for _, secret := range s.Secrets {
+		if contains(inherited, secret.Target) {
+			continue
+		}
+		secretsDir := "/var/run/secrets"
```
Comment (gtardif) [inline local/compose/create.go:303]: That should be`/run/secrets` shouldn't it ?

File: local/compose/create.go (hunk 2 above)
Comment (gtardif) [reply local/compose/create.go:300]: +1 to align with docker-compose behaviour (and indeed I would expect to override with config in case of clash)

**Review by ndeloof — APPROVED** [review]

**Pull request merged**